src/main/java/cafeboard/Post/PostService.java

```diff
@@ -0,0 +1,95 @@
+package cafeboard.Post;
+
+import cafeboard.Board.Board;
+import cafeboard.Board.BoardRepository;
+import cafeboard.Comment.CommentRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.List;
+
+@Service
+public class PostService {
+
+    private final PostRepository postRepository;
+    private final BoardRepository boardRepository;
+    private final CommentRepository commentRepository;
+
+    public PostService(PostRepository postRepository, CommentRepository commentRepository, BoardRepository boardRepository, CommentRepository commentRepository1) {
+        this.postRepository = postRepository;
+        this.boardRepository = boardRepository;
+        this.commentRepository = commentRepository1;
+    }
+
+    //게시글 생성
+    public Post create (CreatePostRequest createRequest){
+        Board board = boardRepository.findById(createRequest.boardId()).orElseThrow(
+                () -> new IllegalArgumentException("존재하지 않는 게시판입니다. 게시글을 생성할 수 없습니다.")
+        );
+
+        Post post = new Post(createRequest.postTitle(),
+                createRequest.postContent(),
+                board);
+
+        return postRepository.save(post);
+    }
+
+
+    //게시글 상세조회(댓글목록 포함)
+    public DetailPostResponse findDetailPost(Long postId) {
+        Post posts = postRepository.findById(postId).orElseThrow(
+                () -> new IllegalArgumentException("해당 게시물이 존재하지 않습니다."));
```

Comment:
이 경우에는 대상 게시물이 없기 때문에 IllegalArgumentException보다 NoSuchElementException를 사용하는 편입니다.
그럼 나중에 IllegalArgumentException인 경우에는 400으로 응답하고 NoSuchElementException인 경우에는 404로 응답할 수 있어요.
```suggestion
                () -> new NoSuchElementException("해당 게시물이 존재하지 않습니다."));
```